Build failure

I tried to follow the instructions in the readme, but got this:

```
$ cmake --build build/DebugNoModules
[21/164] Building CXX object z80/v3/CMakeFiles/z80_v3_make.dir/MakeZ80.cpp.o
FAILED: [code=1] z80/v3/CMakeFiles/z80_v3_make.dir/MakeZ80.cpp.o 
/usr/bin/ccache /usr/lib64/ccache/c++ -DSPECBOLT_EXPORT="" -I/tmp/specbolt -g -std=gnu++26 -fdiagnostics-color=always -Wall -Wextra -pedantic -Wshadow -Wconversion -Werror -MD -MT z80/v3/CMakeFiles/z80_v3_make.dir/MakeZ80.cpp.o -MF z80/v3/CMakeFiles/z80_v3_make.dir/MakeZ80.cpp.o.d -o z80/v3/CMakeFiles/z80_v3_make.dir/MakeZ80.cpp.o -c /tmp/specbolt/z80/v3/MakeZ80.cpp
/tmp/specbolt/z80/v3/MakeZ80.cpp: In constructor ‘constexpr {anonymous}::Opcode::Opcode(std::size_t, const {anonymous}::RegisterSet&)’:
/tmp/specbolt/z80/v3/MakeZ80.cpp:75:74: error: declaration of ‘reg_set’ shadows a member of ‘{anonymous}::Opcode’ [-Werror=shadow]
   75 |   explicit constexpr Opcode(const std::size_t opcode, const RegisterSet &reg_set) :
      |                                                       ~~~~~~~~~~~~~~~~~~~^~~~~~~
/tmp/specbolt/z80/v3/MakeZ80.cpp:73:22: note: shadowed declaration is here
   73 |   const RegisterSet &reg_set;
      |                      ^~~~~~~
```
It's entirely possibly I am doing something very wrong.  I have  little c++ experience, and even less cmake.

--- a/spectrum/include/spectrum/Spectrum.hpp
+++ b/spectrum/include/spectrum/Spectrum.hpp
@@ -175,7 +175,7 @@ private:
   struct TapeTask final : Scheduler::Task {
     Spectrum &spectrum;
     std::size_t last_time_{};
-    explicit TapeTask(Spectrum &spectrum) : spectrum(spectrum) {}
+    explicit TapeTask(Spectrum &spectrum_) : spectrum(spectrum_) {}
     void run(const std::size_t cycles) override {
       spectrum.tape_.pass_time(cycles - last_time_);
       last_time_ = cycles; // ugh

--- a/z80/v3/MakeZ80.cpp
+++ b/z80/v3/MakeZ80.cpp
@@ -72,9 +72,9 @@ struct Opcode {
   std::uint8_t q;
   const RegisterSet &reg_set;
 
-  explicit constexpr Opcode(const std::size_t opcode, const RegisterSet &reg_set) :
+  explicit constexpr Opcode(const std::size_t opcode, const RegisterSet &reg_set_) :
       x(static_cast<std::uint8_t>(opcode) >> 6), y((static_cast<std::uint8_t>(opcode) >> 3) & 0x7),
-      z(static_cast<std::uint8_t>(opcode) & 0x7), p(y >> 1), q(y & 1), reg_set(reg_set) {
+      z(static_cast<std::uint8_t>(opcode) & 0x7), p(y >> 1), q(y & 1), reg_set(reg_set_) {
     if (opcode >= 0x100)
       throw std::runtime_error("Bad opcode");
   }
@@ -356,7 +356,8 @@ Op match_op_ed(const Opcode opcode) {
   if (opcode.x == 1 && opcode.z == 0) {
     return {opcode.y == 6 ? "in (c)"s : std::format("in {}, (c)", opcode.reg_set.r[opcode.y]),
         {"pass_time(4);", // todo IO
-            "const auto result = in(get(R16::BC));", "flags(flags() & Flags::Carry() | Alu::parity_flags_for(result));",
+            "const auto result = in(get(R16::BC));",
+            "flags((flags() & Flags::Carry()) | Alu::parity_flags_for(result));",
             std::format("{}set(R8::{}, result);", opcode.y == 6 ? "//" : "", upper(opcode.reg_set.r[opcode.y]))}};
   }
   if (opcode.x == 1 && opcode.z == 1)
@@ -425,7 +426,7 @@ Op match_op_ed(const Opcode opcode) {
               "set(R8::A, new_a);",
               "pass_time(4);",
               std::format("write(address, static_cast<std::uint8_t>({}));", ind_hl_expression),
-              "flags(flags() & Flags::Carry() | Alu::parity_flags_for(new_a));",
+              "flags((flags() & Flags::Carry()) | Alu::parity_flags_for(new_a));",
           }};
     }
   }

